src/azul/types.py

```diff
@@ -158,6 +164,49 @@ def f(t):
     return tuple(set(f(t)))
 
 
+def get_generic_type_params(cls: type[Generic],
+                            *required_types: type
+                            ) -> Union[list[type], list[TypeVar]]:
```

Comment:
```suggestion
                            ) -> list[Union[type, TypeVar]]:
```

A derived class can leave some of the base class' parameters generic.